[Enhancement] - Added boolean value to enable or disable greeting, ungreeting and trade npc callback

dudantas opened the pull request:
This will allow us to control which npcs will have, by default, the greet, ungreet and trade callbacks enabled, if false, the npc will not use the callback and will need to use something custom
If no parameter (true or false) is provided in the npc NpcHandler:addModule, it will inform the distro with a warning and set the default to true

Added to disable trade callback on "Yana, Peter and Sandomo" npcs (and allow the possibility of modifying in other npcs if necessary)

EnzzoCaaue commented:
don't look good to me bruh

lBaah commented:
Great review @EnzzoCaaue. Lot's of information and data provided as never seen before.
Certainly this is going to change everything.

EnzzoCaaue commented:
yeah but this is to set crit for non black skull players and black skull players, since theyr damage are different

gerotib commented:
As I am testing it doesnt matter at all (if were talking about damage on players with mana shield). With or w/o this commit regular and critical damage on player with & without BS works correct. Looks like this critical applying only matter in CombatHealthFunc and are affecting mana the same way.
Note that I am not using 100% clear repo code now (and cant really do at the moment), but I dont think that I made any change that could affect this.

lBaah commented:
You're totally correct @gerotib. The damage/critical is applied and calculated in health function.
This commit works properly for you?

gerotib commented:
Yeah everything fine

dudantas committed:
Fix some npc trade callbacks

dudantas committed:
Add npcs callbacks to "NpcHandler:addModule()" function

This will allow us to control which npcs will have, by default, the greet, ungreet and trade callbacks enabled, if false, the npc will not use the callback and will need to use something custom

Added to disable trade callback on "Yana, Peter and Sandomo" npcs

Can also be applied to use alternatives to hi or bye

dudantas committed:
Add npc name to addModule to add npc name inside NpcHandler interface

dudantas committed:
Merge branch 'main' into fix-npc-trade-callback

dudantas committed:
Fix conflict

dudantas committed:
Merge branch 'main' into fix-npc-trade-callback
